### PR TITLE
Add checksum verification in ollama.py

### PR DESCRIPTION
**Pull request by swarajpande5**
Adds checksum check after model blobs pulls in ollama.py
Reference: #293 

Example run: 
```
(venv) spande-mac:ramalama (checksum-verification*) $ ./bin/ramalama pull tiny
Warning: podman needs to be configured to use krunkit for AI Workloads,
running without containers

Pulling tinyllama/latest: 100% ▕####################▏ 852/852 2.62MB/s 00:00
File https://registry.ollama.ai/v2/library/tinyllama/blobs/sha256:6331358be52a6ebc2fd0755a51ad1175734fd17a628ab5ea6897109396245362 already fully downloaded.
Pulling 83e42c043d6c7816: 100% ▕####################▏ 608M/608M 3.54MB/s 00:00
Checksum verified for blob /Users/spande/.local/share/ramalama/repos/ollama/blobs/sha256:2af3b81862c6be03c769683af18efdadb2c33f60ff32ab6f83e42c043d6c7816.
```

@ericcurtin If you want checksum verification to be silent (i.e. not printing the "Checksum verified for blob" string), I can amend the commit.

### CODE REVIEW

File: ramalama/ollama.py [hunk 1]
```diff
@@ -87,6 +86,16 @@ def pull_blob(repos, layer_digest, accept, registry_head, models, model_name, mo
...
 
+    # Verify checksum after downloading the blob
+    if not verify_checksum(layer_blob_path):
+        print(f"Checksum mismatch for blob {layer_blob_path}, retrying download...")
+        os.remove(layer_blob_path)
+        download_file(url, layer_blob_path, headers=headers)
+        if not verify_checksum(layer_blob_path):
+            raise ValueError(f"Checksum verification failed for blob {layer_blob_path}")
+    else:
+        print(f"Checksum verified for blob {layer_blob_path}.")
```
Comment (rhatdan) [inline ramalama/ollama.py:97]: This is going to be noisy.  I think humans will just assume what they pulled was correct, and this message will not add anything.

**Comment by ericcurtin** [comment]
Please be silent... But we can have these messages under a debug or verbose option if needs be... Also we only need one progress bar... The model one... The other http request/responses can be silent...

But of course print on errors...

**Comment by swarajpande5** [comment]
> Please be silent... But we can have these messages under a debug or verbose option if needs be... Also we only need one progress bar... The model one... The other http request/responses can be silent...
> 
> But of course print on errors...

@ericcurtin I've observed that 3 separate files are downloaded whenever a model is pulled, two of them are generally really small and the final one is the large one ( which is the model file I assume :) ).
Shall I create a separate issue for this, such that it can be resolved under a separate PR ?

**Comment by ericcurtin** [comment]
> > Please be silent... But we can have these messages under a debug or verbose option if needs be... Also we only need one progress bar... The model one... The other http request/responses can be silent...
> > But of course print on errors...
> 
> @ericcurtin I've observed that 3 separate files are downloaded whenever a model is pulled, two of them are generally really small and the final one is large ( which is the model file I assume :) ). Shall I create a separate issue for this, such that it can be resolved under a separate PR ?

Yeah sure it could be handled in another PR... The second of those files is unnecessary to be downloaded (I had this change in a branch somewhere but maybe I lost that branch)... The first one downloaded doesn't need to be stored in a file, it can just be stored in a variable in python... The only one we need to store is the money

**Comment by swarajpande5** [comment]
@ericcurtin Got it!, I'll start to implement this in a separate PR. Thanks for the suggestions.

**Pull request merged**